pkg/middlewares/route53_records_id_reconcilier.go

```diff
@@ -0,0 +1,49 @@
+package middlewares
+
+import (
+	"strings"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+	"github.com/sirupsen/logrus"
+)
+
+// Since AWS returns the FQDN as the name of the remote record, we must change the Id of the
+// state record to be equivalent (ZoneId_FQDN_Type_SetIdentifier)
+// For a TXT record toto for zone example.com with Id 1234
+// From AWS provider, we retrieve: 1234_toto.example.com_TXT
+// From Terraform state, we retrieve: 1234_toto_TXT
+type Route53RecordIDReconcilier struct{}
+
+func NewRoute53RecordIDReconcilier() Route53RecordIDReconcilier {
+	return Route53RecordIDReconcilier{}
+}
+
+func (m Route53RecordIDReconcilier) Execute(_, resourcesFromState *[]resource.Resource) error {
+
+	for _, remoteResource := range *resourcesFromState {
```

Comment:
Should be `stateResource` instead of `remoteResource`

Reply:
Nice catch thanks